apps/meteor/app/file-upload/server/methods/sendFileMessage.ts

```diff
@@ -22,13 +23,44 @@ import { FileUpload } from '../lib/FileUpload';
 
 function validateFileRequiredFields(file: Partial<IUpload>): asserts file is AtLeast<IUpload, '_id' | 'name' | 'type' | 'size'> {
 	const requiredFields = ['_id', 'name', 'type', 'size'];
-	requiredFields.forEach((field) => {
+	for (const field of requiredFields) {
 		if (!Object.keys(file).includes(field)) {
 			throw new Meteor.Error('error-invalid-file', 'Invalid file');
 		}
-	});
+	}
 }
 
+type parseMultipleFilesIntoMessageAttachmentsResult = { files: FileProp[]; attachments: MessageAttachment[] };
+export const parseMultipleFilesIntoMessageAttachments = async (
+	filesToConfirm: Partial<IUpload>[],
+	roomId: string,
+	user: IUser,
+): Promise<parseMultipleFilesIntoMessageAttachmentsResult> => {
+	const result: parseMultipleFilesIntoMessageAttachmentsResult = { files: [], attachments: [] };
+
+	await Promise.all(
+		filesToConfirm.reduce<Array<Promise<void>>>((acc, file) => {
+			if (!file) return acc;
+
+			acc.push(
+				(async () => {
+					try {
+						const { files, attachments } = await parseFileIntoMessageAttachments(file, roomId, user);
+						result.files.push(...files);
+						result.attachments.push(...attachments);
+					} catch (error) {
+						console.error('Error processing file:', file, error);
+					}
+				})(),
+			);
+
+			return acc;
+		}, []),
+	);
+
+	return result;
```

Comment:
there's no point in using array.reduce if you'll be pushing the promises to the array one by one anyway. Use array.map or a simple for loop instead.

Reply:
@rodrigok FYI

Reply:
@pierre-lehnen-rc  It's a parallel operation, for loop is sequential and map will return a list of promises any ways but will force a second loop to filter results (it was the old implementation)